spec/correcting_scan_helper_spec.rb

```diff
@@ -571,6 +571,83 @@
+              expect(@mock_testcollector).to receive(:testables_tests).and_return(
+                {
+                  'AtomicBoyTests' => [
+                    'AtomicBoyTests/AtomicBoyTests/testExample1',
+                    'AtomicBoyTests/AtomicBoyTests/testExample2',
+                    'AtomicBoyTests/AtomicBoyTests/testExample3',
+                    'AtomicBoyTests/AtomicBoyTests/testExample4'
+                  ]
+                }
+              )
+              expect(scanner).to receive(:correcting_scan)
+                .with(
+                  {
+                    output_directory: '.'
+                  },
+                  anything,
+                  anything
+                )
+              expect(scanner).to receive(:collate_reports)
+              scanner.scan
+            end
+
+            it 'runs only testing with files after failure' do
+              scanner = CorrectingScanHelper.new(
+                xctestrun: 'path/to/fake.xctestrun',
+                output_directory: '.',
+                try_count: 3,
+                quit_simulators: true,
+                invocation_based_tests: true
+              )
+              allow(File).to receive(:exist?).and_call_original
+              allow(File).to receive(:exist?).with(%r{.*/report(-[23])?.junit}).and_return(true)
+              expected_report_files = ['.*/report.junit', '.*/report-2.junit', '.*/report-3.junit']
+              allow(Fastlane::Actions::TestsFromJunitAction).to receive(:run) do |config|
+                expect(config._values).to have_key(:junit)
+                expect(config._values[:junit]).to match(expected_report_files.shift)
+                { failed: ['BagOfTests/CoinTossingUITests/testResultIsTails', 'BagOfTests/CoinTossingUITests/testResultIsTails2'] }
+              end
+              expect(Fastlane::Actions::ScanAction).to receive(:run).ordered.once do |config|
+                expect(config._values).to have_key(:output_files)
+                expect(config._values[:output_files]).to eq('report.html,report.junit')
+                raise FastlaneCore::Interface::FastlaneTestFailure, 'failed tests'
+              end
+              expect(Fastlane::Actions::ScanAction).to receive(:run).ordered.once do |config|
+                expect(config._values).to have_key(:output_files)
+                expect(config._values[:output_files]).to eq('report-2.html,report-2.junit')
+                raise FastlaneCore::Interface::FastlaneTestFailure, 'failed tests'
+              end
+              expect(Fastlane::Actions::ScanAction).to receive(:run).ordered.once do |config|
+                expect(config._values).to have_key(:output_files)
+                expect(config._values[:output_files]).to eq('report-3.html,report-3.junit')
+                expect(config._values).to have_key(:only_testing)
+                expect(config._values[:only_testing]).to eq(['BagOfTests/CoinTossingUITests'])
```

Comment:
> So, when Kiwi tests fail, the junit report will only list the Target/Testsuite or, from what I understand of the code, it will list the Target/Testsuite/TestCase but you cannot pass the full test identifier to :only_testing?

Exactly, because If you tried to run the `Target/Testsuite/TestCase` it not find any the test will conculde with 0 tests run and 0 succeeded